partner-chains-cli/src/prepare_configuration/prepare_chain_params.rs

```diff
@@ -95,12 +93,51 @@ fn is_gov_auth_valid_prompt() -> String {
 	format!("Is the {} displayed above correct?", GOVERNANCE_AUTHORITY.name)
 }
 
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct PaymentKeyFileContent {
+	cbor_hex: String,
+}
+
+const CBOR_KEY_PREFIX: &str = "5820";
+
+fn get_key_hash_from_file(
+	path: &str,
+	context: &impl IOContext,
+) -> anyhow::Result<MainchainAddressHash> {
+	let Some(file_content) = context.read_file(path) else {
+		return Err(anyhow!("Failed to read public key file {path}"));
+	};
+
+	let json = serde_json::from_str::<PaymentKeyFileContent>(&file_content)
+		.map_err(|err| anyhow!("Failed to parse public key file {path}: {err:?}"))?;
+
+	let Some(vkey) = json.cbor_hex.strip_prefix(CBOR_KEY_PREFIX) else {
+		return Err(anyhow!(
+			"Invalid cbor value of public key - missing prefix {CBOR_KEY_PREFIX}: {}",
+			json.cbor_hex
+		));
+	};
+
+	let bytes: [u8; 32] = hex::decode(vkey)
+		.map_err(|err| {
+			anyhow!("Invalid cbor value of public key - not valid hex: {}\n{err:?}", json.cbor_hex)
```

Comment:
does it print the error on the newline or just prints \n? remember having some issues with that

Reply:
It prints a proper newline